open-company/open-company-web
# Refresh read data

Small bug: sometimes the read data fails to load the first time for some problems on the WS connection or other things. If this happens the read data are not never refreshed. This make sure we refresh the WST data for a certain post every time the user opens the WST view. This doesn't distinguish between mobile or desktop, before the refresh was conditional.

Until I discovered this bug the read data was supposed to be always up to date since CS was sending the read data to each connected client every time an update was happening on a given post, but if the read data fails to load the first time we can't be sure they will always be reloaded.

Maybe this approach is a bit too strong, not sure.

To test:
- open a browser tab (open the dev tools before opening the web app to make sure the WS connections are recorded)
- open dev tools and go to network
- navigate to the web app
- now select the WS subgroup of the Network tab of the dev tools
- select the CS connection
- now open the WST for some of the posts you see
- [x] all of them are showing the read data properly? Good
- [x] now open some more and every time you open check that a new `item/who-read` message is sent and a `item/status` message is received
